python/cudf_polars/tests/test_join.py

```diff
@@ -90,20 +110,44 @@ def test_non_coalesce_join(left, right, how, nulls_equal, join_expr):
 def test_coalesce_join(left, right, how, nulls_equal, join_expr):
     if POLARS_VERSION_LT_124:
         query = left.join(
-            right, on=join_expr, how=how, join_nulls=nulls_equal, coalesce=True
+            right,
+            on=join_expr,
+            how=how,
+            join_nulls=nulls_equal,
+            coalesce=True,
+            maintain_order="left_right",
         )
     else:
         query = left.join(
-            right, on=join_expr, how=how, nulls_equal=nulls_equal, coalesce=True
+            right,
+            on=join_expr,
+            how=how,
+            nulls_equal=nulls_equal,
+            coalesce=True,
+            maintain_order="left_right",
```

Comment:
I would prefer to test both the maintain order cases and the case where maintaining order is not required.